tensorflow/dtensor/mlir/tests/dtensor_reduce_scatter_lowering.mlir

```diff
@@ -33,6 +33,48 @@ func.func @lower_reduce_scatter_sum_cpu() -> (tensor<2048x8192xf32>) {
   func.return %3: tensor<2048x8192xf32>
 }
 
+// CHECK-LABEL: func @lower_reduce_scatter_gpu_mesh
+func.func @lower_reduce_scatter_gpu_mesh(%arg0: tensor<i32>,
+           %arg1: tensor<4096x8192xf32> {tf._layout = "sharding_specs:unsharded,unsharded, mesh:gpu_mesh|x=2,y=4|*GPU"}) -> tensor<2048x8192xf32> {
+  // CHECK:      "tf_device.cluster"
+  // CHECK:       %[[DEVICE_ID_RESHAPE:.*]] = "tf.Reshape"(%arg0
+  // CHECK:       %[[RELATIVE_DEVICE_ID:.*]] = "tf.Sub"(%[[DEVICE_ID_RESHAPE]]
+  // CHECK-NEXT:  %[[CONST_1:.*]] = "tf.Const"
+  // CHECK-NEXT:  %[[DEVICE_ID_TO_GROUP_KEY:.*]] = "tf.Const"
+  // CHECK-NEXT:  %[[GROUP_KEY_SLICE:.*]] = "tf.Slice"(%[[DEVICE_ID_TO_GROUP_KEY]], %[[RELATIVE_DEVICE_ID]], %[[CONST_1]]
+  // CHECK:       %[[GROUP_KEY_RESHAPE:.*]] = "tf.Reshape"(%[[GROUP_KEY_SLICE]]
+  // CHECK-NEXT:  %[[INSTANCE_KEY:.*]] = "tf.Const"
+  // CHECK-NEXT:  %[[GROUP_SIZE:.*]] = "tf.Const"
+  // CHECK-NEXT:  %[[REDUCE_SCATTER_OUT:.*]] = "tf.CollectiveReduceScatterV2"(%arg1, %[[GROUP_SIZE]], %[[GROUP_KEY_RESHAPE]], %[[INSTANCE_KEY]])
+  // CHECK-SAME:  final_op = "Id"
+  // CHECK-SAME:  merge_op = "Add"
+  // CHECK-NEXT   tf_device.return %[[REDUCE_SCATTER_OUT]]
+  %0 = "tf_device.cluster"() ({
+    %1 = "tf.Const"() {value = dense<0> : tensor<4x2xi32>} : () -> tensor<4x2xi32>
+    %2 = "tf.Const"() {value = dense<0> : tensor<i32>} : () -> tensor<i32>
```

Comment:
Is this const for the GroupAssignment argument? If so, I think it shall be updated to a 2d vector to conform to the usual expectation, like this[1]. This is likely causing this mlir test to fail at an assertion line in our internal version. (The difference could be explained if the usual OSS tests didn't enable these assertions?).

[1] https://github.com/tensorflow/tensorflow/blob/master/tensorflow/dtensor/mlir/tests/dtensor_allreduce_lowering.mlir#L12

Reply:
There is another similar use in the other test a few lines below.

Reply:
Actually, the const at line 53 is the problematic one. It shouldn't be all zeros, but actually an array of unique 'device_ids'.

Reply:
Thanks, I updated all of the tests in this file.